### ISSUE TITLE
Spacelift stack must match ID if imported

### ISSUE THREAD

**Issue by lexton** [spacelift-io/terraform-provider-spacelift#315]
## Steps to re-produce

* Create resource manually in UI
* Import the spacelift stack using terraform
* In the terraform resource specify the slug

## Definition
``` terraform
resource "spacelift_stack" "this" {
  github_enterprise {
    namespace = var.github_org
  }

  slug           = "spacelift-seed"
  name           = "Spacelift Admin"
  administrative = true
  autodeploy     = true
  autoretry      = false
  branch         = var.branch
  labels = [
    "folder:team/${var.team}/admin",
    "folder:env/${var.env}",
    "folder:worker/${var.worker_pool_name}",
  ]
  github_action_deploy  = false
  enable_local_preview  = true
  manage_state          = true
  project_root          = var.root
  protect_from_deletion = true
  repository            = var.repository
  terraform_version     = var.terraform_version
  worker_pool_id        = var.worker_pool_id

}
```

## plan output
```
  # module.spacelift-admin-stack.spacelift_stack.this must be replaced
-/+ resource "spacelift_stack" "this" {
      ...
      ~ id                               = "spacelift-seed" -> (known after apply)
        name                             = "Spacelift Admin"
      + slug                             = "spacelift-seed" # forces replacement
        # (13 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }
```

## workaround

Rename stack to make id match its name, and remove the slug field. You can't add the slug field back after the fact - and resources are unable to be imported to match the slug.

## Expected behavior

It shouldn't need to re-build if the slug matches the expected ID since we actually already know the ID prior to the plan.

**Comment by lexton** [comment]
Shouldn't the slug converge to the same field as the ID? This wouldn't be unknown if it is set in the slug value. 

I'm using slug to mean 'configurable but static id'. 

I don't think the example you provided works as it would generate a slug 'spacelift-admin' rather than 'spacelift-seed' as far as I can tell. 

In the case where it isn't configurable - I wouldn't want the import to be accepted if the slug is set. There isn't really a world where I would want to import only to have terraform want to rebuild it.

**Comment by adamconnelly** [comment]
> Shouldn't the slug converge to the same field as the ID?

Yes, the slug is the ID. Initially the provider didn't allow setting a custom slug, so the slug field was added after the fact.

> I'm using slug to mean 'configurable but static id'.

That's fairly accurate. The only thing I'd stress is that the slug can only be set once, at the time of Stack creation. It can't be set after the fact.

> I don't think the example you provided works as it would generate a slug 'spacelift-admin' rather than 'spacelift-seed' as far as I can tell.

My assumption here was that you were importing an existing stack with the slug `spacelift-seed`. Maybe I misunderstood, so here's the rough steps I used:

1. Create a new stack via the UI called "Spacelift Seed".
2. Rename that stack to "Spacelift Admin". At this point the slug will be "spacelift-seed", but the name will be "Spacelift Admin".
3. Create your `spacelift_stack` resource in Terraform, but don't specify the `slug` property.
4. Import the stack via terraform import "spacelift_stack.this" "spacelift-seed".

At this point if you run a `terraform plan` it should show no changes.

> In the case where it isn't configurable - I wouldn't want the import to be accepted if the slug is set. There isn't really a world where I would want to import only to have terraform want to rebuild it.

That's fair. I think we've got at least two options at this point:

1. Require the `slug` property when importing existing stacks.
2. Return an error if the `slug` is specified when importing a stack. This would keep the current behaviour of it being write-once, but would prevent an accidental recreate.

**Comment by adamconnelly** [comment]
@lexton quick update - I took another look at this and realised that there was a simple solution. We just needed to mark the `slug` as computed, and that prevents Terraform trying to recreate stacks in the scenario where someone hasn't specified the slug in their configuration.

Long story short when the change is released it should fix your original issue without you having to remove slug from your configuration.
